source/common/buffer/buffer_impl.h

```diff
@@ -52,17 +48,27 @@ class Slice {
   Slice() = default;
 
   /**
-   * Create an empty mutable Slice that owns its storage, which it charges to
-   * the provided account, if any.
+   * Create an empty mutable Slice that owns its storage, which it charges to the provided account,
+   * if any.
    * @param min_capacity number of bytes of space the slice should have. Actual capacity is rounded
    * up to the next multiple of 4kb.
    * @param account the account to charge.
-   * @param freelist to search for the backing storage, if any.
    */
-  Slice(uint64_t min_capacity, BufferMemoryAccountSharedPtr account,
-        absl::optional<FreeListReference> free_list = absl::nullopt)
-      : capacity_(sliceSize(min_capacity)), storage_(newStorage(capacity_, free_list)),
-        base_(storage_.get()), data_(0), reservable_(0) {
+  Slice(uint64_t min_capacity, const BufferMemoryAccountSharedPtr& account)
+      : Slice(newStorage(min_capacity), 0, account) {}
+
+  /**
+   * Create an empty mutable Slice that owns its storage, which it charges to the provided account,
+   * if any.
+   * @param storage backend storage for the slice.
+   * @param size the size already used in storage.
+   * @param account the account to charge.
+   */
+  Slice(SizedStorage storage, uint64_t size, const BufferMemoryAccountSharedPtr& account)
+      : capacity_(storage.len_), storage_(std::move(storage.mem_)), base_(storage_.get()),
```

Comment:
the `capacity_` seems a copy of `storage_.len_`

Reply:
yes.

Reply:
oops, mess up with the `stroage_`'s type, apologize for the invalid review comment.